ast/compile.go

```diff
@@ -1151,6 +1185,16 @@ func parseSchema(schema interface{}) (types.Type, error) {
 	return types.A, nil
 }
 
+func (c *Compiler) setAnnotationSet() {
+	// Recursion is caught in earlier step, so this cannot fail.
+	sorted, _ := c.Graph.Sort()
+	as, errs := buildAnnotationSet(sorted)
+	for _, err := range errs {
+		c.err(err)
+	}
+	c.annotationSet = as
```

Comment:
I had been staring at this for a minute. Since `buildAnnotationSet` returns `nil, errs` when there are errors, this is resetting c.annotationSet to `nil` when there are errors. I guess that's intended.  Also fits in with the surrounding code... ✔️